src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicy.cs

```diff
@@ -141,5 +149,23 @@ public override string ToString()
             builder.Append("}");
             return builder.ToString();
         }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="CorsPolicy"/>.
+        /// </summary>
+        public CorsPolicy()
+        {
+            // Regenerate 'RegularOrigins' every time the 'Origins' has changed.
+            Origins.CollectionChanged += (sender, e) =>
```

Comment:
Are policies all generated and set during configuration? If so using a `ObservableCollection` seems to be an overkill. Are there mechanism to update patterns once when all the origins are added?